meteor/client/ui/Shelf/ShelfContextMenu.tsx

```diff
@@ -145,10 +147,17 @@ export default function ShelfContextMenu(props: ShelfContextMenuProps) {
 						</MenuItem>
 					)}
 				</>
-			)
+			) : null
 		}
 	}
 
+	const startExecuteMenuItems =
+		context?.type === ContextType.ADLIB
```

Comment:
Be carefull with nesting ternaries. They become real difficult to read real quick.
This here some more like a use-case for a switch since you are checking on `context.type` multiple times.

I see that you just moved the ternaries. Consider refactoring it to a switch :)